app/views.py

```diff
@@ -72,16 +72,18 @@ def get(self, request, game_id, player_id):
         player = Player.players.get(pk=player_id)
 
         thumbs_seen, eyes_seen = player_info(game, player)
+        
+        roles = ', '.join(list(map(lambda x: str(x.role), game.player_set.filter(is_kicked=False).all())))
```

Comment:
does django have a better way to do `map` things

Reply:
this is how i normally do it in plain old python but its ugly af imo

Reply:
`', '.join([p.role if not p.is_kicked for p in game.player_set])` ? 
idk syntax exactly tho

Reply:
I think you want `roles = game.players().order_by('role').values_list('role', flat=True)` and then `<p>{{ roles|join:", " }}</p>` in the template.

Reply:
i agree with what you said tristan, the joining should be template level

will do today

Reply:
what's the difference between `.players` and `.player_set` btw?

Reply:
oh this is a django query set? You can tack on a `.filter(is_kicked=False)` somewhere

Reply:
b i k e s h e d b o y s